pkg/rbd/nodeserver.go

```diff
@@ -303,15 +303,32 @@ func (ns *NodeServer) mountVolumeToStagePath(ctx context.Context, req *csi.NodeS
 	// Publish Path
 	fsType := req.GetVolumeCapability().GetMount().GetFsType()
 	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: mount.NewOsExec()}
+	existingFormat, err := diskMounter.GetDiskFormat(devicePath)
+	if err != nil {
+		return err
+	}
+	if existingFormat == "" && (fsType == "ext4" || fsType == "xfs") {
+		args := []string{devicePath}
+		if fsType == "ext4" {
+			args = []string{"-F", "-m0", "-E nodiscard", devicePath}
```

Comment:
Does `-E nodiscard` work for you? If I paste that in as a single argument on my command line, I get an error from `mkfs.ex4`

Reply:
@dillaman I enabled the ext4 specific test with `-Enodiscard` and its passing. so, it works!